crypto/rsa/rsa_backend.c

```diff
@@ -96,25 +109,73 @@ int ossl_rsa_fromdata(RSA *rsa, const OSSL_PARAM params[], int include_private)
                                 ossl_rsa_mp_coeff_names))
             goto err;
 
-        /* It's ok if this private key just has n, e and d */
-        if (sk_BIGNUM_num(factors) != 0
-            && !ossl_rsa_set0_all_params(rsa, factors, exps, coeffs))
-            goto err;
+        if (derive_from_pq && (sk_BIGNUM_num(exps) == 0) && (sk_BIGNUM_num(coeffs) == 0)) {
```

Comment:
What if exps are provided but not coeffs or vice versa?

Reply:
In this case I would just overwrite the provided ones. I.e., there should be `||` instead of `&&`

Reply:
I thought in the OTC meeting we agreed that overwriting was not something we wanted to do in this path?  i.e. if items were provided that didn't allow for the derivation of missing parameters, we would either raise an error or allow an incomplete key to be caught by a later call to EVP_PKEY_check

Reply:
Returning an error is also an option. Ignoring the derive_from_pq param is IMO not acceptable.